active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,894 @@
+when the public CreateIconElement method is called.
+
+_Spec note: this naming pattern of a public Foo and protected virtual FooCore is an
+existing pattern in Xaml today. The other name suffix that's used is "Override", but
+"Core" is more frequent._
+
+## NavigationViewItem.Icon
+
+_Spec note: this isn't a new property, but has new behavior_
+
+If you set an AnimatedIcon as the value of the Icon property, 
+the NavigationViewItem will set the states of the AnimatedIcon for you, according to the states of the control.
+
+For example, this sets a custom animation `PlayIcon` that was generated by the LottieGen tool:
+
+```xml
+<NavigationView.MenuItems>
+    <NavigationViewItem Content = "Play">
+        <NavigationViewItem.Icon>
+
+            <AnimatedIcon >
+                <local:PlayIcon />
+            </AnimatedIcon>
+
+    </NavigationViewItem.Icon>
+</NavigationView.MenuItems>
+```
+
+The NavigationViewItem will automatically set the following states on the AnimatedIcon:
+* Normal
+* PointerOver
+* Pressed
+* Selected
+* PressedSelected
+* PointerOverSelected
+
+If the PlayIcon has a marker for "NormalToPointerOver", it will be animated
+to when the mouse moves to over the NavigationViewItem.
+See [link] for more information on combining marker names.
+
+To create this custom PlayIcon class, you run a Lottie JSON file (with markers) through the Windows
+[CodeGen](https://docs.microsoft.com/en-us/windows/communitytoolkit/animations/lottie-scenarios/getting_started_codegen)
+tool to generate the animation code as a C# class.
+
+Once you run your Lottifile through codegen you can add the output class to your project. 
+Follow the steps in the
+[CodeGen](https://docs.microsoft.com/en-us/windows/communitytoolkit/animations/lottie-scenarios/getting_started_codegen)
+tutorial for more instructions. 
+
+## IAnimatedVisualSource2 interface
+
+Provides a Composition `Visual` that can be animated.
+An object that implements this interface can be used as the Source property of an AnimatedIcon.
+
+Call the TryCreateAnimatedVisual method to retrieve an
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual).
+`IAnimatedVisual` can then be used to get a
+Composition [Visual](https://docs.microsoft.com/uwp/api/Windows.UI.Composition.Visual),
+which can be added to a XAML element tree using the methods of
+[ElementCompositionPreview](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Hosting.ElementCompositionPreview).
+
+Use the `SetColorProperty` method to set a color of the animated visual,
+and the `Markers` property to find named positions in the animation timeline.
+
+_Spec note: `IAnimatedVisualSource2` is like `IAnimatedVisualSource` in purpose and use,
+but there is no derivation because the '2' interface does not derive from
+[IDynamicAnimatedVisualSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IDynamicAnimatedVisualSource)._
```

Comment:
How did I**Dynamic**AnimatedVisualSource enter the picture? The 2 interface would presumably derive from IAnimatedVisualSource.

Or are you saying that IDynamicAnimatedVisualSource does not derive from IAnimatedVisualSource2? Would we expect it to?

Reply:
Discussion: I got confused and reversed the inheritance order.

Recommend: delete that